### PR TITLE
OMERO server Ansible role

### PR DESCRIPTION
**Pull request by manics**
An Ansible role for installing and upgrading OMERO.server (uses omego). Needs more testing, everyone's welcome to give it an initial check.

For a basic example see `ansible/roles/omero-server/README.md`.

For a more advanced example see `openstack/examples/` which contains scripts and playbooks for provisioning a new OpenStack VM, and for deploying an all-in-one OMERO.server/postgres on it. Alternatively you should be able to run `openstack/examples/idr-example-omero-bootstrap.sh` directly on a CentOS7 system.

- [ ] TODO: Some paths are currently hard-coded to this fork/branch so that they can be tested

### CODE REVIEW

**Comment by joshmoore** [comment]
Generally, :thumbsup:. A few questions:

 * would a generalization of https://github.com/openmicroscopy/infrastructure/pull/45 et al to not be jenkins-based support installing omego with the need for a whole role (which seems fairly anemic)?
 * worth merging https://github.com/openmicroscopy/infrastructure/pull/58 and then adding a vagrant box for testing this?
 * what's the first target production system for this?

**Comment by manics** [comment]
> would a generalization of #45 et al to not be jenkins-based support installing omego with the need for a whole role (which seems fairly anemic)?

Maybe, though it's probably easier to just include `omego` in this role instead.

> worth merging #58 and then adding a vagrant box for testing this?

Yes

> what's the first target production system for this?

IDR, hence the openstack example

**Comment by joshmoore** [comment]
Note: my primary question regarding this strategy is what's the benefit over, say, the previous docker strategy. As I see it there are now (minimally) these choices:

 * bare-metal / follow-docs (most common, I'd think)
 * bare-metal / omero-install
 * bare-metal / omego
 * bare-metal / ansible
 * docker / omero-install
 * docker / omego
 * docker / ansible
 * docker / omero-grid (I was assuming this was our target)

**Comment by joshmoore** [comment]
Running bootstrap.sh in an omero-ssh-systemd docker:

```
TASK [omero-server : omero web | selinux booleans] *****************************
failed: [localhost] => (item=httpd_read_user_content) => {"failed": true, "item": "httpd_read_user_content", "msg": "This module requires libsemanage-python support"}
failed: [localhost] => (item=httpd_enable_homedirs) => {"failed": true, "item": "httpd_enable_homedirs", "msg": "This module requires libsemanage-python support"}
```

File: openstack/examples/idr-example-omero.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    postgresql_users_databases:
+    - user: omero
+      password: omero
+      databases: [omero]
+    omero_dbhost: localhost
+    omero_dbuser: omero
+    omero_dbname: omero
+    omero_dbpassword: omero
+    omero_datadir: /OMERO
+    omero_selinux_setup: True
```
Comment (joshmoore) [inline openstack/examples/idr-example-omero.yml]: Running bootstrap.sh in an omero-ssh-systemd docker:

```
TASK [omero-server : omero web | selinux booleans] *****************************
failed: [localhost] => (item=httpd_read_user_content) => {"failed": true, "item": "httpd_read_user_content", "msg": "This module requires libsemanage-python support"}
failed: [localhost] => (item=httpd_enable_homedirs) => {"failed": true, "item": "httpd_enable_homedirs", "msg": "This module requires libsemanage-python support"}
```

File: openstack/examples/idr-example-omero-bootstrap.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/opt/ansible/bin/pip -q install ansible 2>&1
+
+cd /opt
+
+git clone -q --single-branch -b omego https://github.com/manics/infrastructure.git /opt/infrastructure 2>&1
+
+# Custom variable overrides (YAML, can be empty)
+cat << EOF > /opt/infrastructure/localhost-extravars.yml
+omero_release: "5.2.2"
+#omero_release: OMERO-build
```
Comment (joshmoore) [inline openstack/examples/idr-example-omero-bootstrap.sh]: Using `OMERO-DEV-merge-build` didn't work for me. Trying with `--build https://...` now

Comment (joshmoore) [inline openstack/examples/idr-example-omero-bootstrap.sh]: I've had to add it to the extra_vars line rather than the extravars.yml. Unsure why.

**Comment by joshmoore** [comment]
Running the delete playbook I get this:

```
TASK [Remove OMERO external access security group] *****************************
task path: /Users/jamoore/opt/management_tools/infrastructure/openstack/examples/idr-os-playbook-delete.yml:18
...
"Error deleting security group ...: Security Group ... in use."
```

Expected?

**Comment by manics** [comment]
Yes, if there are other instances using the same security groups you can't delete them. I wrote `idr-os-playbook-delete.yml` for convenience whilst testing- should I delete it?

**Comment by joshmoore** [comment]
As long as it's not generally dangerous, I think it's fine. Just want to understand. My general feeling is that we should get all of these merged so that we have a single view and then we can start cleaning/expanding/explaining.

**Comment by manics** [comment]
@joshmoore
- Merged https://github.com/manics/infrastructure/pull/1
- switched to using ansible from EPEL since it's now at v2
- gone back to using the vars file (`+omero_release: OMERO-DEV-merge-build`)
- Autodetect selinux

**Comment by manics** [comment]
Since ` openstack/examples/omedev-docker.yml` is included as an example in this role it now depends on #67 (Or if it's easier I could move `omedev-docker.yml` into a separate PR).

**Comment by joshmoore** [comment]
Can we get https://github.com/openmicroscopy/infrastructure/pull/61/commits/23d4bf0ae87eb5c545fcfa8012dbf7b5acf288b0 merged and then move https://github.com/openmicroscopy/infrastructure/pull/61/commits/92ffc96ea64a7e3c25d49d2d4eac58e8c6d77c54 to gh-67?

**Comment by joshmoore** [comment]
(doh; this didn't get posted):
 > I retested [this](https://github.com/openmicroscopy/infrastructure/pull/61/commits/23d4bf0ae87eb5c545fcfa8012dbf7b5acf288b0) to create `idrstaging`. Works fine. Happy for this to go in and we can work on refactoring later.

**Comment by manics** [comment]
Rebased onto master, removed commits related to `openstack/examples/omedev-docker.yml`, force-pushed

**Comment by joshmoore** [comment]
Merging along with https://github.com/IDR/idr-metadata/pull/82 as a tidy up of a previous tested state. This will need further work to clean out hard-coded references that will no longer exist.

**Pull request merged**